sources/doublemap/route.rb

```diff
@@ -11,11 +11,17 @@ def refresh
         # Ensure that the short name of the route (commonly it's identifier) is set.
         attrs[:short_name]  = route.name.split.first if route.short_name.empty?
         attrs[:itinerary]    = route.stops.map do |stop_id|
-          Shark::Station.identifier_for(@station_key.call(api.stops.get(stop_id)))
-        end
+          station_id = @station_key.call(api.stops.get(stop_id))
+          Shark::Station.identifier_for(station_id) if station_id
+        end.compact
         ["routes."+@route_key.call(route), attrs]
       end.to_h
     end
+
+    def valid?(route)
+      @route_key.call(route) and
```

Comment:
General styling prefers `&&` over `and` because of differences in precedence and the fact that `and` does _not_ short circuit. Particularly with a `.all` as the right operand, it seems like it might be better to use `&&` here.